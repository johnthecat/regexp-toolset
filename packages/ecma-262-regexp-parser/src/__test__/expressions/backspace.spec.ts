import { expect, describe, it } from 'vitest';
import { parseRegexp } from '../../index.js';

describe('Backspace', () => {
  it('should correctly parse backspace', () => {
    expect(parseRegexp(/[\b]/)).toMatchSnapshot();
  });

  it('should correctly work, surronded by other characters', () => {
    expect(parseRegexp(/[a\bc]/)).toMatchSnapshot();
  });
});
