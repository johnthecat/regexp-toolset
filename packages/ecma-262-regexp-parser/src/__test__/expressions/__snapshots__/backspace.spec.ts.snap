// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Backspace > should correctly parse backspace 1`] = `
{
  "body": {
    "end": 4,
    "expressions": [
      {
        "end": 3,
        "kind": 10,
        "start": 2,
      },
    ],
    "kind": 21,
    "negative": false,
    "start": 1,
  },
  "end": 5,
  "flags": "",
  "kind": 0,
  "start": 0,
}
`;

exports[`Backspace > should correctly work, surronded by other characters 1`] = `
{
  "body": {
    "end": 6,
    "expressions": [
      {
        "charCode": 97,
        "end": 2,
        "kind": 6,
        "start": 2,
        "type": "simple",
        "value": "a",
      },
      {
        "end": 4,
        "kind": 10,
        "start": 3,
      },
      {
        "charCode": 99,
        "end": 5,
        "kind": 6,
        "start": 5,
        "type": "simple",
        "value": "c",
      },
    ],
    "kind": 21,
    "negative": false,
    "start": 1,
  },
  "end": 7,
  "flags": "",
  "kind": 0,
  "start": 0,
}
`;
